src/metabase/api/search.clj

```diff
@@ -345,25 +370,63 @@
                              (map #(update % :favorite bit->boolean))
                              (map #(update % :archived bit->boolean))
                              (map (partial scoring/score-and-result (:search-string search-ctx)))
-                             (filter some?))]
-      (scoring/top-results reducible-results xf))))
-
+                             (filter some?))
+          total-results     (scoring/top-results reducible-results xf)]
+      ;; We get to do this slicing and dicing with the result data because
+      ;; the pagination of search is for UI improvement, not for performance.
+      ;; We intend for the cardinality of the search results to be below the default max before this slicing occurs
+      {
+       :total (count total-results)
+       :data
+       (cond->> total-results
+         (some? (:offset-int search-ctx)) (drop (:offset-int search-ctx))
+         (some? (:limit-int search-ctx)) (take (:limit-int search-ctx)))
+       :limit (:limit-int search-ctx)
+       :offset (:offset-int search-ctx)
+       :table_db_id (:table-db-id search-ctx)
+       :models (:models search-ctx) })))
```

Comment:
nits: I don't like it, but the opening brace should be on the same line as the first keyword. Also, it's nice if all the values are vertically aligned (I use the `align-cljlet` package in Emacs)